crates/cheatcodes/src/evm.rs

```diff
@@ -695,6 +782,129 @@ fn inner_delete_state_snapshots<DB: DatabaseExt>(ccx: &mut CheatsCtxt<DB>) -> Re
     Ok(Default::default())
 }
 
+fn inner_value_snapshot<DB: DatabaseExt>(
+    ccx: &mut CheatsCtxt<DB>,
+    group: Option<String>,
+    name: Option<String>,
+    value: String,
+) -> Result {
+    let (group, name) = derive_snapshot_name(ccx, group, name);
+
+    ccx.state.gas_snapshots.entry(group).or_default().insert(name, value);
+
+    Ok(Default::default())
+}
+
+fn inner_last_gas_snapshot<DB: DatabaseExt>(
+    ccx: &mut CheatsCtxt<DB>,
+    group: Option<String>,
+    name: Option<String>,
+    value: u64,
+) -> Result {
+    let (group, name) = derive_snapshot_name(ccx, group, name);
+
+    ccx.state.gas_snapshots.entry(group).or_default().insert(name, value.to_string());
+
+    Ok(value.abi_encode())
+}
+
+fn inner_start_gas_snapshot<DB: DatabaseExt>(
+    ccx: &mut CheatsCtxt<DB>,
+    group: Option<String>,
+    name: Option<String>,
+) -> Result {
+    // Revert if there is an active gas snapshot as we can only have one active snapshot at a time.
+    if ccx.state.gas_metering.active_gas_snapshot.is_some() {
+        let (group, name) = ccx.state.gas_metering.active_gas_snapshot.as_ref().unwrap().clone();
+        bail!("gas snapshot was already started with group: {group} and name: {name}");
+    }
+
+    let group = group.as_deref().unwrap_or(
+        ccx.state.config.running_contract.as_deref().expect("expected running contract"),
+    );
+    let name = name.as_deref().unwrap_or("default").to_string();
```

Comment:
since we're passing String already we can avoid the additional to string here if we make default.to_string the else